sdk/core/azure-core/src/main/java/com/azure/core/implementation/ResponseConstructorsCache.java

```diff
@@ -65,7 +64,8 @@
                 try {
                     return (Constructor<? extends Response<?>>) constructor;
                 } catch (Throwable t) {
-                    throw logger.logExceptionAsError(new RuntimeException(t));
+                    throw logger.logExceptionAsError(new RuntimeException("Failed to cast the constructor to Response."
+                        + "Error Details: " + t.getMessage(), t));
```

Comment:
Do you need `"Error Details:" + t.getMessage`? In previous log messages you didn't have this, and you've already passed the inner exception.

Reply:
I want to expose the detail message from inner exception to upper layer. Otherwise it will show up a little under. Hard to trace.

Reply:
After a second thought. It might be useful to revert it back to origin one.

Reply:
It's a problem with our ClientLogger that should be resolved. Unless we explicitly say in the spec that we should be logging Throwable.getMessage().